toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Helpers.jsm

```diff
@@ -16,48 +16,90 @@ class ArrayBufferDataStream {
         return rv;
     }
 
+    writeUint8(value) {
+        this.dataView.setUint8(this.pos, value);
+        this.pos += 1;
+    }
+
     readUint16() {
         let rv = this.dataView.getUint16(this.pos);
         this.pos += 2;
         return rv;
     }
 
+    writeUint16(value) {
+        this.dataView.setUint16(this.pos, value);
+        this.pos += 2;
+    }
+
     readUint32() {
         let rv = this.dataView.getUint32(this.pos);
         this.pos += 4;
         return rv;
     }
 
+    writeUint32(value) {
+        this.dataView.setUint32(this.pos, value);
+        this.pos += 4;
+    }
+
     readUint64() {
-        let rv = this.dataView.getUint64(this.pos);
+        let rv = this.dataView.getBigUint64(this.pos);
```

Comment:
there is no `getUint64` in `dataView`.. it only supports writing `BigInt` in 64 bits. So I convert back and forth between `Number` and `BigInt`

Reply:
Should we keep this as a `BigInt` and not convert to `Number`?  The reason I ask is the issue I mentioned on slack with Numbers that need > 60 bits or so.  I think this would mean also updating the scaffolding functions to use `bigint` for objects.